backend/addcorpus/models.py

```diff
@@ -46,6 +60,66 @@ class Meta:
     def __str__(self):
         return self.name
 
+    @admin.display()
+    def ready_to_index(self) -> bool:
+        '''
+        Checks whether the corpus is ready for indexing.
+
+        Runs a try/except around `self.validate_ready_to_index()` and returns a
+        boolean; `True` means the validation completed without errors.
+
+        If you want to see validation error messages, use the validation method
+        directly.
+        '''
+        return _validation_passes(self.validate_ready_to_index)
+
+    def validate_ready_to_index(self) -> None:
+        '''
+        Validation that should be carried out before indexing.
+
+        Raises:
+            CorpusNotIndexableError: the corpus is not meeting requirements for making
+                an index.
+        '''
+
+        validate_has_configuration(self)
+
+        config = self.configuration
+        fields = config.fields.all()
+
+        validate_essential_fields(fields)
+
+
+    @admin.display()
+    def ready_to_publish(self) -> bool:
+        '''
+        Checks whether the corpus is ready to be made public.
+        '''
+        return _validation_passes(self.validate_ready_to_publish)
```

Comment:
Tying in to the comment above, I would place the `try/except` here, and return `True/False`